File: csharp/Ice/hello/HelloI.cs
// Copyright (c) ZeroC, Inc.

using Demo;

internal class HelloI : HelloDisp_
{
    public override void sayHello(int delay, Ice.Current current)
    {
        if (delay > 0)
        {
            System.Threading.Thread.Sleep(delay);
        }
        Console.Out.WriteLine("Hello World!");
    }

    public override void shutdown(Ice.Current current)
    {
        Console.Out.WriteLine("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }
}

File: java/Ice/hello/HelloI.java
// Copyright (c) ZeroC, Inc.

import com.zeroc.demos.Ice.hello.Demo.*;

public class HelloI implements Hello
{
    @Override
    public void sayHello(int delay, com.zeroc.Ice.Current current)
    {
        if(delay > 0)
        {
            try
            {
                Thread.currentThread();
                Thread.sleep(delay);
            }
            catch(InterruptedException ex1)
            {
            }
        }
        System.out.println("Hello World!");
    }

    @Override
    public void shutdown(com.zeroc.Ice.Current current)
    {
        System.out.println("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }
}
